### PR TITLE
[16.0][ADD] contract_analytic_tag: New module

### PR DESCRIPTION
**Pull request by victoralmau**
New module

Please @pedrobaeza and @carlos-lopez-tecnativa can you review it?

@Tecnativa TT51221

### CODE REVIEW

**Comment by carlos-lopez-tecnativa** [comment]
I think it's necessary to adapt to the new analytic distribution model instead of replicating the older one using analytic tags. Am I missing any context?

**Comment by pedrobaeza** [comment]
In this case, we need this module because the analytic tag is used for classifying analytic lines, not for doing distributions, so the new model doesn't work for this (nor the manual distribution).

**Review by carlos-lopez-tecnativa — APPROVED** [review]
@pedrobaeza thanks for clarifying. 
Tested, LGTM

File: contract_analytic_tag/models/contract_line.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    analytic_tag_ids = fields.Many2many(
+        comodel_name="account.analytic.tag",
+        string="Analytic Tags",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+    )
+
+    def _prepare_invoice_line(self):
+        vals = super()._prepare_invoice_line()
+        if self.analytic_tag_ids:
+            vals.update({"analytic_tag_ids": [(6, 0, self.analytic_tag_ids.ids)]})
```
Comment (pedrobaeza) [inline contract_analytic_tag/models/contract_line.py:18]: Simpler syntax:

```suggestion
            vals["analytic_tag_ids"] = [(6, 0, self.analytic_tag_ids.ids)]
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
This should go to OCA/contract

**Comment by victoralmau** [comment]
Moved to https://github.com/OCA/contract/pull/1133

**Pull request closed without merging**